clang/lib/Parse/ParseStmt.cpp

```diff
@@ -2421,6 +2426,52 @@ StmtResult Parser::ParseBreakStatement() {
   return Actions.ActOnBreakStmt(BreakLoc, getCurScope());
 }
 
+/// ParseContractAssertStatement
+///
+///  assertion-statement:
+///     'contract_assert' attribute-specifier-seq[opt] '('
+///     conditional-expression ')' ';'
+///
+StmtResult Parser::ParseContractAssertStatement() {
+  assert((Tok.is(tok::kw_contract_assert)) &&
+         "Not a contract asssert statement");
+  SourceLocation KeywordLoc = ConsumeToken(); // eat the 'contract_assert'.
+
+  // FIXME(EricWF): This seems really worg.
+  ParsedAttributes attrs(AttrFactory);
+  MaybeParseCXX11Attributes(attrs);
```

Comment:
looks fine to me? (except we never do anything with these attributes)
We might want to make ContractStmt an AttributedStmt